internal/experiment/openvpn/openvpn.go

```diff
@@ -313,15 +172,29 @@ func (m *Measurer) connectAndHandshake(
 				Auth:        openvpnConfig.OpenVPNOptions().Auth,
 				Compression: string(openvpnConfig.OpenVPNOptions().Compress),
 			},
-			T0:            tFirst,
-			T:             tLast,
+			T0:            t0,
+			T:             t,
 			Tags:          []string{},
 			TransactionID: index,
 		},
 		NetworkEvents: handshakeEvents,
 	}
 }
 
+func TimestampsFromHandshake(events []*vpntracex.Event) (float64, float64, float64) {
+	var (
+		t0       float64
+		t        float64
+		duration float64
+	)
+	if len(events) > 0 {
+		t0 = events[0].AtTime
+		t = events[len(events)-1].AtTime
+		duration = t - t0
+	}
```

Comment:
I think how it is now it will be a bit more precise.

For UDP the difference is not very big (first event is the PRE_START event, i.e. when we initialize the state machine):

```
 302         "t0": 0.000184251, // to as-is, i.e. first event
 303         "t00": 0.0000224, // t00 as before call
```
// TODO(https://github.com/ooni/probe/issues/2755): make the code that fetches experiment private
// and let the common code export just the bare minimum to make this possible.


For TCP however I think it's more informative because it discounts the TCP 3-way handshake:

```

1270           "cipher": "AES-256-GCM"
1271         },
1272         "t0": 0.040491564, // to as-is, first event
1273         "t00": 0.00001647, // t00 as in before call
```

in case of error, well, I think it's good to have the hardcoded timeout showing one way or another (statistically it should be visible as a ceiling?)

Reply:
>  Does it make sense to define the bootstrap time when there is no actual finishing bootstrap event? 

it's got `omitempty` set, so in case of a zero-duration interval the key is not serialized.

Reply:
We have discussed this topic with @ainghazal in a call that we had today. We're not going to perform this work as part of this pull request but rather as a subsequent pull request, where we'll try to harmonize with the way in which other experiments based on the `tunnel` specification work. @ainghazal can you please create a placeholder issue for that such that we remember to do this work before the merge window closes?

Reply:
https://github.com/ooni/probe/issues/2758

Reply:
Here we go: https://github.com/ooni/probe/issues/2758